src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,157 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private int size;
+    private Node<T>[] items;
+
+    public ArrayList() {
+        items = new Node[MAX_ARRAY_LENGTH];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < items.length) {
+            items[size] = new Node<>(value);
+            size++;
+        } else {
+            growArray(size);
+            items[size] = new Node<>(value);
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        } else if (index == size) {
+            add(value);
+        } else {
+            if (size + 1 <= items.length) {
+                addInside(index, value);
+            } else {
+                growArray(size);
+                addInside(index, value);
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        if (getIndexInRange(index)) {
+            return items[index].getValue();
+        }
         return null;
```

Comment:
it's optional to have `if` statement, because in method getIndexInRange() you throw exception, so it should work without `if` statement
```suggestion
        getIndexInRange(index);
        return items[index].getValue();
```